[v3]layerscape: add 64b/32b target for ls1043ardb device

fsl-jyt opened the pull request:
Add support for NXP layerscape ls1043ardb 64b/32b Dev board.

LS1043a is an SoC with 4x64-bit up to 1.6 GHz ARMv8 A53 cores.
ls1043ardb support features as: 2GB DDR4, 128MB NOR/512MB NAND, USB3.0, eSDHC,
I2C, GPIO, PCIe/Mini-PCIe, 6x1G/1x10G network port, etc.

64b/32b ls1043ardb target is using 4.4 kernel, and rcw/u-boot/fman images from
NXP QorIQ SDK release.

All of 4.4 kernel patches porting from SDK release or upstream.

QorIQ SDK ISOs can be downloaded from this location:
http://www.nxp.com/products/software-and-tools/run-time-software/linux-sdk/linux-sdk-for-qoriq-processors:SDKLINUX

Signed-off-by: Yutang Jiang yutang.jiang@nxp.com

fsl-jyt committed:
layerscape: add 64b/32b target for ls1043ardb device

Add support for NXP layerscape ls1043ardb 64b/32b Dev board.

LS1043a is an SoC with 4x64-bit up to 1.6 GHz ARMv8 A53 cores.
ls1043ardb support features as: 2GB DDR4, 128MB NOR/512MB NAND, USB3.0, eSDHC,
I2C, GPIO, PCIe/Mini-PCIe, 6x1G/1x10G network port, etc.

64b/32b ls1043ardb target is using 4.4 kernel, and rcw/u-boot/fman images from
NXP QorIQ SDK release.

All of 4.4 kernel patches porting from SDK release or upstream.

QorIQ SDK ISOs can be downloaded from this location:
http://www.nxp.com/products/software-and-tools/run-time-software/linux-sdk/linux-sdk-for-qoriq-processors:SDKLINUX

Signed-off-by: Yutang Jiang <yutang.jiang@nxp.com>

fsl-jyt commented:
In sync with https://github.com/lede-project/source/pull/302

V3 patch update summary:
 1. Optimize and minimize the config
 2. Using PACKAGES/kmod mechanism
 3. Add i2c-imx in i2c.mk and sdhci-of-esdhc in other.mk

wigyori commented:
@fsl-jyt - could you please verify if it compiles and runs fine with 4.4.14 which is currently in OpenWrt trunk? Same goes for the image building.

Thanks,
Zoltan H

fsl-jyt commented:
Yes, Zoltan H. I have compiled both 32b and 64b in currently OpenWrt 4.4.14 kernel, image building also ok, and validated the mainly features.

Please have a look the kernel bootup log(64b):

U-Boot 2016.012.0+ga9b437f (May 15 2016 - 11:01:10 +0800)

SoC:  unknown (0x87920410)
Clock Configuration:
       CPU0(A53):1600 MHz  CPU1(A53):1600 MHz  CPU2(A53):1600 MHz
       CPU3(A53):1600 MHz
       Bus:      400  MHz  DDR:      1600 MT/s  FMAN:     500  MHz
Reset Configuration Word (RCW):
       00000000: 08100010 0a000000 00000000 00000000
       00000010: 14550002 80004012 e0025000 c1002000
       00000020: 00000000 00000000 00000000 00038800
       00000030: 00000000 00001101 00000096 00000001
I2C:   ready
Model: LS1043A RDB Board
Board: LS1043ARDB, boot from vBank 0
CPLD:  V1.4
PCBA:  V2.0
SERDES Reference Clocks:
SD1_CLK1 = 156.25MHZ, SD1_CLK2 = 100.00MHZ
DRAM:  Initializing DDR....
Detected UDIMM Fixed DDR on board
2 GiB (DDR4, 32-bit, CL=11, ECC off)
SEC0: RNG instantiated
fsl-ppa: Bad firmware image (not a FIT image)
fsl-ppa: error (-22)
Not a microcode
fsl-ppa: Bad firmware image (not a FIT image)
Waking secondary cores to start from ffd0a000
All (4) cores are up.
Using SERDES1 Protocol: 5205 (0x1455)
Flash: 128 MiB
NAND:  512 MiB
MMC:   FSL_SDHC: 0
EEPROM: NXID v1
PCIe1: disabled
PCIe2: Root Complex x1 gen1, regs @ 0x3500000
PCI:
     01:00.0    - 168c:0030 - Network controller
PCIe2: Bus 00 - 01
PCIe3: Root Complex no link, regs @ 0x3600000
In:    serial
Out:   serial
Err:   serial
SCSI:  Error: SCSI Controller(s) 1B4B:9170 not found
Net:   Fman1: Uploading microcode version 106.4.18
FM1@DTSEC1, FM1@DTSEC2, FM1@DTSEC3, FM1@DTSEC4, FM1@DTSEC5, FM1@DTSEC6, FM1@TGEC1
Hit any key to stop autoboot:  0
=> pri
baudrate=115200
bootargs=console=ttyS0,115200 root=/dev/ram0 earlycon=uart8250,0x21c0500,115200
bootcmd=cpld reset altbank
bootdelay=1
console=ttyAMA0,38400n8
eth1addr=00:e0:0c:00:77:01
eth2addr=00:e0:0c:00:77:02
eth3addr=00:e0:0c:00:77:03
eth4addr=00:e0:0c:00:77:04
eth5addr=00:e0:0c:00:77:05
eth6addr=00:e0:0c:00:77:06
ethact=FM1@DTSEC1
ethaddr=00:e0:0c:00:77:00
ethprime=e1000#0
fdt_high=0xffffffffffffffff
fdtaddr=8f000000
fdtcontroladdr=ffbc7738
fman_ucode=60300000
gatewayip=192.168.1.1
hwconfig=fsl_ddr:bank_intlv=auto
initrd_high=0xffffffffffffffff
ipaddr=10.193.20.234
kernel_addr=0x100000
kernel_load=0xa0000000
kernel_size=0x2800000
kernel_start=0x61100000
loadaddr=82000000
netmask=255.255.255.0
ramdisk_addr=0x800000
ramdisk_size=0x2000000
ramdiskaddr=88000000
scsidevs=0
serverip=10.192.208.142
stderr=serial
stdin=serial
stdout=serial
wrtboot=setenv bootargs root=/dev/mtdblock6 rootfstype=squashfs,jffs2 noinitrd earlycon=uart8250,mmio,0x21c0500 console=ttyS0,115200 mtdparts=60000000.nor:1M(rcw),1M(u-boot),1M(u-boot-env),1M(fman),1M(dtb),5M(kernel),54M(rootfs),64M(otherbank),60M@0x400000(firmware) && cp.b 60400000 $fdtaddr 100000 && cp.b 60500000 $loadaddr 500000 && bootm $loadaddr - $fdtaddr

Environment size: 1289/131068 bytes
=> run wrtboot
## Booting kernel from Legacy Image at 82000000 ...
   Image Name:   ARM64 OpenWrt Linux-4.4.14
   Image Type:   AArch64 Linux Kernel Image (gzip compressed)
   Data Size:    3238887 Bytes = 3.1 MiB
   Load Address: 80080000
   Entry Point:  80080000
   Verifying Checksum ... OK
## Flattened Device Tree blob at 8f000000
   Booting using the fdt blob at 0x8f000000
   Uncompressing Kernel Image ... OK
   Using Device Tree in place at 000000008f000000, end 000000008f0197c6
fsl-ppa: Bad firmware image (not a FIT image)
fsl-ppa: Bad firmware image (not a FIT image)

Starting kernel ...

[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 4.4.14 (jyt@BP) (gcc version 5.3.0 (OpenWrt GCC 5.3.0 49930) ) #1 SMP Fri Sep 2 17:08:07 UTC 2016
[    0.000000] Boot CPU: AArch64 Processor [410fd034]
[    0.000000] earlycon: Early serial console at MMIO 0x21c0500 (options '')
[    0.000000] bootconsole [uart0] enabled
[    0.000000] efi: Getting EFI parameters from FDT:
[    0.000000] efi: UEFI not found.
[    0.000000] Reserved memory: initialized node qman-fqd, compatible id fsl,qman-fqd
[    0.000000] Reserved memory: initialized node qman-pfdr, compatible id fsl,qman-pfdr
[    0.000000] Reserved memory: initialized node bman-fbpr, compatible id fsl,bman-fbpr
[    0.000000] PERCPU: Embedded 15 pages/cpu @ffffffc07fd8e000 s23808 r8192 d29440 u61440
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] CPU features: enabling workaround for ARM erratum 845719
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 515592
[    0.000000] Kernel command line: root=/dev/mtdblock6 rootfstype=squashfs,jffs2 noinitrd earlycon=uart8250,mmio,0x21c0500 console=ttyS0,115200 mtdparts=60000000.nor:1M(rcw),1M(u-boot),1M(u-boot-env),1M(fman),1M(dtb),5M(kernel),54M(rootfs),64M(otherbank),60M@0x400000(firmware)
[    0.000000] PID hash table entries: 4096 (order: 3, 32768 bytes)
[    0.000000] Dentry cache hash table entries: 262144 (order: 9, 2097152 bytes)
[    0.000000] Inode-cache hash table entries: 131072 (order: 8, 1048576 bytes)
[    0.000000] software IO TLB [mem 0xf6200000-0xfa200000] (64MB) mapped at [ffffffc076200000-ffffffc07a1fffff]
[    0.000000] Memory: 1927068K/2095104K available (4781K kernel code, 282K rwdata, 2120K rodata, 264K init, 811K bss, 168036K reserved, 0K cma-reserved)
[    0.000000] Virtual kernel memory layout:
[    0.000000]     vmalloc : 0xffffff8000000000 - 0xffffffbdbfff0000   (   246 GB)
[    0.000000]     vmemmap : 0xffffffbdc0000000 - 0xffffffbfc0000000   (     8 GB maximum)
[    0.000000]               0xffffffbdc0000000 - 0xffffffbdc1ff8000   (    31 MB actual)
[    0.000000]     fixed   : 0xffffffbffabfd000 - 0xffffffbffac00000   (    12 KB)
[    0.000000]     PCI I/O : 0xffffffbffae00000 - 0xffffffbffbe00000   (    16 MB)
[    0.000000]     modules : 0xffffffbffc000000 - 0xffffffc000000000   (    64 MB)
[    0.000000]     memory  : 0xffffffc000000000 - 0xffffffc07fe00000   (  2046 MB)
[    0.000000]       .init : 0xffffffc000740000 - 0xffffffc000782000   (   264 KB)
[    0.000000]       .text : 0xffffffc000080000 - 0xffffffc00073f4e4   (  6910 KB)
[    0.000000]       .data : 0xffffffc000792000 - 0xffffffc0007d8a00   (   283 KB)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] Hierarchical RCU implementation.
[    0.000000]  CONFIG_RCU_FANOUT set to non-default value of 32
[    0.000000] NR_IRQS:64 nr_irqs:64 0
[    0.000000] GIC: Using split EOI/Deactivate mode
[    0.000000] Architected cp15 timer(s) running at 25.00MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x5c40939b5, max_idle_ns: 440795202646 ns
[    0.000002] sched_clock: 56 bits at 25MHz, resolution 40ns, wraps every 4398046511100ns
[    0.008164] Calibrating delay loop (skipped), value calculated using timer frequency.. 50.00 BogoMIPS (lpj=250000)
[    0.018568] pid_max: default: 32768 minimum: 301
[    0.023237] Mount-cache hash table entries: 4096 (order: 3, 32768 bytes)
[    0.029972] Mountpoint-cache hash table entries: 4096 (order: 3, 32768 bytes)
[    0.037518] EFI services will not be available.
[    0.042083] ASID allocator initialised with 65536 entries
[    0.048012] Detected VIPT I-cache on CPU1
[    0.048030] CPU1: Booted secondary processor [410fd034]
[    0.048145] Detected VIPT I-cache on CPU2
[    0.048157] CPU2: Booted secondary processor [410fd034]
[    0.048265] Detected VIPT I-cache on CPU3
[    0.048277] CPU3: Booted secondary processor [410fd034]
[    0.048297] Brought up 4 CPUs
[    0.079067] SMP: Total of 4 processors activated.
[    0.083789] CPU: All CPU(s) started at EL2
[    0.087906] alternatives: patching kernel code
[    0.095133] DMI not present or invalid.
[    0.099088] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.109031] atomic64_test: passed
[    0.112721] NET: Registered protocol family 16
[    0.117574] fsl-mc bus type registered
[    0.121398] MC object device driver fsl_mc_dprc registered
[    0.126933] MC object device driver fsl_mc_allocator registered
[    0.132922] Bman ver:0a02,02,01
[    0.140386] qman-fqd addr 0xff000000 size 0x800000
[    0.145201] qman-pfdr addr 0xfc000000 size 0x2000000
[    0.150186] Qman ver:0a01,03,02,00
[    0.153672] vdso: 2 pages (1 code @ ffffffc000799000, 1 data @ ffffffc000798000)
[    0.161207] DMA: preallocated 256 KiB pool for atomic allocations
[    0.167364] Serial: AMBA PL011 UART driver
[    0.211862] ACPI: Interpreter disabled.
[    0.215938] SCSI subsystem initialized
[    0.219920] usbcore: registered new interface driver usbfs
[    0.225477] usbcore: registered new interface driver hub
[    0.230837] usbcore: registered new device driver usb
[    0.235982] dmi: Firmware registration failed.
[    0.240469] bman-fbpr addr 0xfe000000 size 0x1000000
[    0.245481] Bman err interrupt handler present
[    0.250351] Bman portal initialised, cpu 0
[    0.254522] Bman portal initialised, cpu 1
[    0.258695] Bman portal initialised, cpu 2
[    0.262870] Bman portal initialised, cpu 3
[    0.266978] Bman portals initialised
[    0.271432] Qman err interrupt handler present
[    0.276191] QMan: Allocated lookup table at ffffff800016d000, entry count 131073
[    0.284359] Qman portal initialised, cpu 0
[    0.288926] Qman portal initialised, cpu 1
[    0.293492] Qman portal initialised, cpu 2
[    0.298053] Qman portal initialised, cpu 3
[    0.302167] Qman portals initialised
[    0.305842] Bman: BPID allocator includes range 32:32
[    0.310951] Qman: FQID allocator includes range 256:256
[    0.316200] Qman: FQID allocator includes range 32768:32768
[    0.321837] Qman: CGRID allocator includes range 0:256
[    0.327115] Qman: pool channel allocator includes range 1025:15
[    0.333116] No USDPAA memory, no 'fsl,usdpaa-mem' in device-tree
[    0.339187] fsl-ifc 1530000.ifc: Freescale Integrated Flash Controller
[    0.345749] fsl-ifc 1530000.ifc: IFC version 1.4, 8 banks
[    0.351737] clocksource: Switched to clocksource arch_sys_counter
[    0.358002] pnp: PnP ACPI: disabled
[    0.362076] NET: Registered protocol family 2
[    0.366695] TCP established hash table entries: 16384 (order: 5, 131072 bytes)
[    0.374054] TCP bind hash table entries: 16384 (order: 6, 262144 bytes)
[    0.380883] TCP: Hash tables configured (established 16384 bind 16384)
[    0.387496] UDP hash table entries: 1024 (order: 3, 32768 bytes)
[    0.393561] UDP-Lite hash table entries: 1024 (order: 3, 32768 bytes)
[    0.400148] NET: Registered protocol family 1
[    0.404649] RPC: Registered named UNIX socket transport module.
[    0.410592] RPC: Registered udp transport module.
[    0.415318] RPC: Registered tcp transport module.
[    0.420038] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    0.427019] futex hash table entries: 1024 (order: 5, 131072 bytes)
[    0.436692] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.442935] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.452986] fuse init (API version 7.23)
[    0.457370] io scheduler noop registered (default)
[    0.462780] PCI host bridge /soc/pcie@3500000 ranges:
[    0.467860]    IO 0x4800010000..0x480001ffff -> 0x00000000
[    0.473376]   MEM 0x4840000000..0x487fffffff -> 0x40000000
[    0.478967] layerscape-pcie 3500000.pcie: PCI host bridge to bus 0000:00
[    0.485706] pci_bus 0000:00: root bus resource [bus 00-ff]
[    0.491214] pci_bus 0000:00: root bus resource [io  0x0000-0xffff]
[    0.497425] pci_bus 0000:00: root bus resource [mem 0x4840000000-0x487fffffff] (bus address [0x40000000-0x7fffffff])
[    0.521788] pci 0000:00:00.0: BAR 1: assigned [mem 0x4840000000-0x4843ffffff]
[    0.528958] pci 0000:00:00.0: BAR 0: assigned [mem 0x4844000000-0x4844ffffff]
[    0.536136] pci 0000:00:00.0: BAR 6: assigned [mem 0x4845000000-0x4845ffffff pref]
[    0.543746] pci 0000:00:00.0: BAR 8: assigned [mem 0x4846000000-0x48460fffff]
[    0.550916] pci 0000:01:00.0: BAR 0: assigned [mem 0x4846000000-0x484601ffff 64bit]
[    0.558649] pci 0000:01:00.0: BAR 6: assigned [mem 0x4846020000-0x484602ffff pref]
[    0.566258] pci 0000:00:00.0: PCI bridge to [bus 01]
[    0.571243] pci 0000:00:00.0:   bridge window [mem 0x4846000000-0x48460fffff]
[    0.578577] PCI host bridge /soc/pcie@3600000 ranges:
[    0.583658]    IO 0x5000010000..0x500001ffff -> 0x00000000
[    0.589167]   MEM 0x5040000000..0x507fffffff -> 0x40000000
[    0.594737] layerscape-pcie 3600000.pcie: PCI host bridge to bus 0001:00
[    0.601469] pci_bus 0001:00: root bus resource [bus 00-ff]
[    0.606984] pci_bus 0001:00: root bus resource [io  0x10000-0x1ffff] (bus address [0x0000-0xffff])
[    0.615992] pci_bus 0001:00: root bus resource [mem 0x5040000000-0x507fffffff] (bus address [0x40000000-0x7fffffff])
[    0.626709] pci 0001:00:00.0: bridge configuration invalid ([bus 00-00]), reconfiguring
[    0.634808] pci 0001:00:00.0: BAR 1: assigned [mem 0x5040000000-0x5043ffffff]
[    0.641983] pci 0001:00:00.0: BAR 0: assigned [mem 0x5044000000-0x5044ffffff]
[    0.649152] pci 0001:00:00.0: BAR 6: assigned [mem 0x5045000000-0x5045ffffff pref]
[    0.656761] pci 0001:00:00.0: PCI bridge to [bus 01]
[    0.661766] pcieport 0001:00:00.0: enabling device (0000 -> 0002)
[    0.668312] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    0.675133] console [ttyS0] disabled
[    0.678734] 21c0500.serial: ttyS0 at MMIO 0x21c0500 (irq = 18, base_baud = 25000000) is a 16550A
[    0.687579] console [ttyS0] enabled
[    0.687579] console [ttyS0] enabled
[    0.694556] bootconsole [uart0] disabled
[    0.694556] bootconsole [uart0] disabled
[    0.702514] 21c0600.serial: ttyS1 at MMIO 0x21c0600 (irq = 18, base_baud = 25000000) is a 16550A
[    0.711367] of_serial: probe of 21d0500.serial failed with error -28
[    0.717746] of_serial: probe of 21d0600.serial failed with error -28
[    0.724352] Unable to detect cache hierarchy from DT for CPU 0
[    0.733483] brd: module loaded
[    0.738529] loop: module loaded
[    0.743213] 60000000.nor: Found 1 x16 devices at 0x0 in 16-bit bank. Manufacturer ID 0x000089 Chip ID 0x00227e
[    0.753208] Amd/Fujitsu Extended Query Table at 0x0040
[    0.758343]   Amd/Fujitsu Extended Query version 1.3.
[    0.763389] number of CFI chips: 1
[    0.766790] 9 cmdlinepart partitions found on MTD device 60000000.nor
[    0.773222] Creating 9 MTD partitions on "60000000.nor":
[    0.778527] 0x000000000000-0x000000100000 : "rcw"
[    0.783423] ftl_cs: FTL header not found.
[    0.787621] 0x000000100000-0x000000200000 : "u-boot"
[    0.792759] ftl_cs: FTL header not found.
[    0.796937] 0x000000200000-0x000000300000 : "u-boot-env"
[    0.802424] ftl_cs: FTL header not found.
[    0.806617] 0x000000300000-0x000000400000 : "fman"
[    0.811572] ftl_cs: FTL header not found.
[    0.815757] 0x000000400000-0x000000500000 : "dtb"
[    0.820625] ftl_cs: FTL header not found.
[    0.824808] 0x000000500000-0x000000a00000 : "kernel"
[    0.829937] ftl_cs: FTL header not found.
[    0.834126] 0x000000a00000-0x000004000000 : "rootfs"
[    0.839257] ftl_cs: FTL header not found.
[    0.843436] mtd: device 6 (rootfs) set to be root filesystem
[    0.849096] 1 squashfs-split partitions found on MTD device rootfs
[    0.855271] 0x000000d80000-0x000004000000 : "rootfs_data"
[    0.860834] ftl_cs: FTL header not found.
[    0.865016] 0x000004000000-0x000008000000 : "otherbank"
[    0.870409] ftl_cs: FTL header not found.
[    0.874593] 0x000000400000-0x000004000000 : "firmware"
[    0.879896] ftl_cs: FTL header not found.
[    0.884614] nand: device found, Manufacturer ID: 0x2c, Chip ID: 0xac
[    0.890956] nand: Micron MT29F4G08ABBDAH4
[    0.894959] nand: 512 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    0.902783] Bad block table found at page 262080, version 0x01
[    0.909077] Bad block table found at page 262016, version 0x01
[    0.917207] ftl_cs: FTL header not found.
[    0.921391] fsl,ifc-nand 7e800000.nand: IFC NAND device at 0x7e800000, bank 1
[    0.928959] libphy: Fixed MDIO Bus: probed
[    0.933158] libphy: Freescale XGMAC MDIO Bus: probed
[    0.939449] libphy: Freescale XGMAC MDIO Bus: probed
[    0.945933] libphy: Freescale XGMAC MDIO Bus: probed
[    0.950931] libphy: Freescale XGMAC MDIO Bus: probed
[    0.955935] libphy: Freescale XGMAC MDIO Bus: probed
[    0.960931] libphy: Freescale XGMAC MDIO Bus: probed
[    0.965934] libphy: Freescale XGMAC MDIO Bus: probed
[    0.970934] libphy: Freescale XGMAC MDIO Bus: probed
[    0.975935] libphy: Freescale XGMAC MDIO Bus: probed
[    0.992425] Freescale FM module, FMD API version 21.1.0
[    0.999331] Freescale FM Ports module
[    1.002992] fsl_mac: fsl_mac: FSL FMan MAC API based driver
[    1.008897] fsl_mac 1ae0000.ethernet: FMan MEMAC
[    1.013515] fsl_mac 1ae0000.ethernet: FMan MAC address: 00:e0:0c:00:77:00
[    1.020580] fsl_mac 1ae2000.ethernet: FMan MEMAC
[    1.025195] fsl_mac 1ae2000.ethernet: FMan MAC address: 00:e0:0c:00:77:01
[    1.032023] fsl_mac 1ae4000.ethernet: FMan MEMAC
[    1.036632] fsl_mac 1ae4000.ethernet: FMan MAC address: 00:e0:0c:00:77:02
[    1.043462] fsl_mac 1ae6000.ethernet: FMan MEMAC
[    1.048072] fsl_mac 1ae6000.ethernet: FMan MAC address: 00:e0:0c:00:77:03
[    1.055143] fsl_mac 1ae8000.ethernet: FMan MEMAC
[    1.059752] fsl_mac 1ae8000.ethernet: FMan MAC address: 00:e0:0c:00:77:04
[    1.066823] fsl_mac 1aea000.ethernet: FMan MEMAC
[    1.071433] fsl_mac 1aea000.ethernet: FMan MAC address: 00:e0:0c:00:77:05
[    1.078266] fsl_mac 1af0000.ethernet: FMan MEMAC
[    1.082881] fsl_mac 1af0000.ethernet: FMan MAC address: 00:e0:0c:00:77:06
[    1.089735] fsl_dpa: FSL DPAA Ethernet driver
[    1.095812] fsl_dpa: fsl_dpa: Probed interface eth0
[    1.102195] fsl_dpa: fsl_dpa: Probed interface eth1
[    1.108920] fsl_dpa: fsl_dpa: Probed interface eth2
[    1.116015] fsl_dpa: fsl_dpa: Probed interface eth3
[    1.123475] fsl_dpa: fsl_dpa: Probed interface eth4
[    1.131286] fsl_dpa: fsl_dpa: Probed interface eth5
[    1.139470] fsl_dpa: fsl_dpa: Probed interface eth6
[    1.144400] fsl_advanced: FSL DPAA Advanced drivers:
[    1.149355] fsl_proxy: FSL DPAA Proxy initialization driver
[    1.155045] fsl_dpa_shared: FSL DPAA Shared Ethernet driver
[    1.160693] fsl_dpa_macless: FSL DPAA MACless Ethernet driver
[    1.166517] fsl_oh: FSL FMan Offline Parsing port driver
[    1.172657] Freescale USDPAA process driver
[    1.176829] fsl-usdpaa: no region found
[    1.180656] Freescale USDPAA process IRQ driver
[    1.185546] NET: Registered protocol family 10
[    1.190644] NET: Registered protocol family 17
[    1.195101] bridge: automatic filtering via arp/ip/ip6tables has been deprecated. Update your scripts to load br_netfilter if you need this.
[    1.207700] 8021q: 802.1Q VLAN Support v1.8
[    1.212242] fsl_generic: FSL DPAA Generic Ethernet driver
[    1.218208] fdt: not creating '/sys/firmware/fdt': CRC check failed
[    1.226374] VFS: Mounted root (squashfs filesystem) readonly on device 31:6.
[    1.233532] Freeing unused kernel memory: 264K (ffffffc000740000 - ffffffc000782000)
[    1.241275] Freeing alternatives memory: 44K (ffffffc000782000 - ffffffc00078d000)
[    1.588799] init: Console is alive
[    1.713986] pps_core: LinuxPPS API ver. 1 registered
[    1.718947] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    1.728407] PTP clock support registered
[    2.334736] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    2.340221] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 1
[    2.347954] xhci-hcd xhci-hcd.0.auto: hcc params 0x0220f66d hci version 0x100 quirks 0x00010010
[    2.356670] xhci-hcd xhci-hcd.0.auto: irq 27, io mem 0x02f00000
[    2.362800] hub 1-0:1.0: USB hub found
[    2.366549] hub 1-0:1.0: 1 port detected
[    2.370544] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    2.376029] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 2
[    2.383719] usb usb2: We don't know the algorithms for LPM for this host, disabling LPM.
[    2.391974] hub 2-0:1.0: USB hub found
[    2.395721] hub 2-0:1.0: 1 port detected
[    2.399729] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    2.405216] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 3
[    2.412937] xhci-hcd xhci-hcd.1.auto: hcc params 0x0220f66d hci version 0x100 quirks 0x00010010
[    2.421639] xhci-hcd xhci-hcd.1.auto: irq 28, io mem 0x03000000
[    2.427731] hub 3-0:1.0: USB hub found
[    2.431477] hub 3-0:1.0: 1 port detected
[    2.435472] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    2.440950] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 4
[    2.448623] usb usb4: We don't know the algorithms for LPM for this host, disabling LPM.
[    2.456874] hub 4-0:1.0: USB hub found
[    2.460621] hub 4-0:1.0: 1 port detected
[    2.464654] xhci-hcd xhci-hcd.2.auto: xHCI Host Controller
[    2.470135] xhci-hcd xhci-hcd.2.auto: new USB bus registered, assigned bus number 5
[    2.477856] xhci-hcd xhci-hcd.2.auto: hcc params 0x0220f66d hci version 0x100 quirks 0x00010010
[    2.486563] xhci-hcd xhci-hcd.2.auto: irq 29, io mem 0x03100000
[    2.492667] hub 5-0:1.0: USB hub found
[    2.496414] hub 5-0:1.0: 1 port detected
[    2.500405] xhci-hcd xhci-hcd.2.auto: xHCI Host Controller
[    2.505890] xhci-hcd xhci-hcd.2.auto: new USB bus registered, assigned bus number 6
[    2.513562] usb usb6: We don't know the algorithms for LPM for this host, disabling LPM.
[    2.521821] hub 6-0:1.0: USB hub found
[    2.525568] hub 6-0:1.0: 1 port detected
[    2.531036] sdhci: Secure Digital Host Controller Interface driver
[    2.537217] sdhci: Copyright(c) Pierre Ossman
[    2.541773] sdhci-pltfm: SDHCI platform and OF driver helper
[    2.548025] usbcore: registered new interface driver usb-storage
[    2.555690] sdhci-esdhc 1560000.esdhc: No vmmc regulator found
[    2.561511] sdhci-esdhc 1560000.esdhc: No vqmmc regulator found
[    2.611744] mmc0: SDHCI controller on 1560000.esdhc [1560000.esdhc] using ADMA 64-bit
[    2.624455] init: - preinit -
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    2.811756] usb usb5-port1: over-current condition
[    2.841750] usb usb6-port1: over-current condition
[    5.743965] jffs2: notice: (850) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
[    5.759779] mount_root: switching to jffs2 overlay
[    5.767657] procd: - early -
[    6.382413] procd: - ubus -
[    6.434799] random: ubusd urandom read with 4 bits of entropy available
[    6.441685] procd: - init -
Please press Enter to activate this console.
[    6.508755] i2c /dev entries driver
[    6.512984] i2c i2c-0: IMX I2C adapter registered
[    6.517680] i2c i2c-0: can't use DMA
[    6.522801] e1000e: Intel(R) PRO/1000 Network Driver - 3.2.6-k
[    6.528625] e1000e: Copyright(c) 1999 - 2015 Intel Corporation.
[    7.992255] device eth0 entered promiscuous mode
[    7.997580] br-lan: port 1(eth0) entered forwarding state
[    8.002988] br-lan: port 1(eth0) entered forwarding state
[    9.992204] br-lan: port 1(eth0) entered disabled state



BusyBox v1.24.2 () built-in shell (ash)

  _______                     ________        __
|       |.-----.-----.-----.|  |  |  |.----.|  |_
|   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
|_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
-----------------------------------------------------
DESIGNATED DRIVER (Bleeding Edge, 49929)
-----------------------------------------------------
  * 2 oz. Orange Juice         Combine all juices in a
  * 2 oz. Pineapple Juice      tall glass filled with
  * 2 oz. Grapefruit Juice     ice, stir well.
  * 2 oz. Cranberry Juice
-----------------------------------------------------
root@OpenWrt:/# ifconfig
br-lan    Link encap:Ethernet  HWaddr 00:E0:0C:00:77:00
          inet addr:192.168.1.1  Bcast:192.168.1.255  Mask:255.255.255.0
          inet6 addr: fd67:8579:a0a8::1/60 Scope:Global
          inet6 addr: fe80::2e0:cff:fe00:7700/64 Scope:Link
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:8 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0 (0.0 B)  TX bytes:896 (896.0 B)

eth0      Link encap:Ethernet  HWaddr 00:E0:0C:00:77:00
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:9 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0 (0.0 B)  TX bytes:942 (942.0 B)
          Memory:1ae0000-1ae0fff

eth1      Link encap:Ethernet  HWaddr 00:E0:0C:00:77:01
          inet addr:10.193.20.43  Bcast:10.193.20.255  Mask:255.255.255.0
          inet6 addr: fe80::2e0:cff:fe00:7701/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:49 errors:0 dropped:0 overruns:0 frame:0
          TX packets:10 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:4172 (4.0 KiB)  TX bytes:1844 (1.8 KiB)
          Memory:1ae2000-1ae2fff

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:8 errors:0 dropped:0 overruns:0 frame:0
          TX packets:8 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:656 (656.0 B)  TX bytes:656 (656.0 B)

root@OpenWrt:/# ping 10.193.20.106
PING 10.193.20.106 (10.193.20.106): 56 data bytes
64 bytes from 10.193.20.106: seq=0 ttl=64 time=12.728 ms
64 bytes from 10.193.20.106: seq=1 ttl=64 time=0.169 ms
^C
--- 10.193.20.106 ping statistics ---
2 packets transmitted, 2 packets received, 0% packet loss
round-trip min/avg/max = 0.169/6.448/12.728 ms
root@OpenWrt:/#

From: Zoltan Herpai [mailto:notifications@github.com]
Sent: Saturday, September 03, 2016 4:40 PM
To: openwrt/openwrt
Cc: Yutang Jiang; Mention
Subject: Re: [openwrt/openwrt] [v3]layerscape: add 64b/32b target for ls1043ardb device (#81)


@fsl-jyt<https://github.com/fsl-jyt> - could you please verify if it compiles and runs fine with 4.4.14 which is currently in OpenWrt trunk? Same goes for the image building.

Thanks,
Zoltan H

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/openwrt/openwrt/pull/81#issuecomment-244534852>, or mute the thread<https://github.com/notifications/unsubscribe-auth/APKzf-Mj4BpuCF1o1mU5HO9UZwXUtzFTks5qmTJ3gaJpZM4JyW3Q>.

fsl-jyt commented:
I notice of other target have private modules.mk, I will also place those kmod- to modules.mk in next patch version.

From: Yutang Jiang
Sent: Monday, September 05, 2016 10:27 AM
To: 'openwrt/openwrt'; openwrt/openwrt
Cc: Mention
Subject: RE: [openwrt/openwrt] [v3]layerscape: add 64b/32b target for ls1043ardb device (#81)

Yes, Zoltan H. I have compiled both 32b and 64b in currently OpenWrt 4.4.14 kernel, image building also ok, and validated the mainly features.

Please have a look the kernel bootup log(64b):

U-Boot 2016.012.0+ga9b437f (May 15 2016 - 11:01:10 +0800)

SoC:  unknown (0x87920410)
Clock Configuration:
       CPU0(A53):1600 MHz  CPU1(A53):1600 MHz  CPU2(A53):1600 MHz
       CPU3(A53):1600 MHz
       Bus:      400  MHz  DDR:      1600 MT/s  FMAN:     500  MHz
Reset Configuration Word (RCW):
       00000000: 08100010 0a000000 00000000 00000000
       00000010: 14550002 80004012 e0025000 c1002000
       00000020: 00000000 00000000 00000000 00038800
       00000030: 00000000 00001101 00000096 00000001
I2C:   ready
Model: LS1043A RDB Board
Board: LS1043ARDB, boot from vBank 0
CPLD:  V1.4
PCBA:  V2.0
SERDES Reference Clocks:
SD1_CLK1 = 156.25MHZ, SD1_CLK2 = 100.00MHZ
DRAM:  Initializing DDR....
Detected UDIMM Fixed DDR on board
2 GiB (DDR4, 32-bit, CL=11, ECC off)
SEC0: RNG instantiated
fsl-ppa: Bad firmware image (not a FIT image)
fsl-ppa: error (-22)
Not a microcode
fsl-ppa: Bad firmware image (not a FIT image)
Waking secondary cores to start from ffd0a000
All (4) cores are up.
Using SERDES1 Protocol: 5205 (0x1455)
Flash: 128 MiB
NAND:  512 MiB
MMC:   FSL_SDHC: 0
EEPROM: NXID v1
PCIe1: disabled
PCIe2: Root Complex x1 gen1, regs @ 0x3500000
PCI:
     01:00.0    - 168c:0030 - Network controller
PCIe2: Bus 00 - 01
PCIe3: Root Complex no link, regs @ 0x3600000
In:    serial
Out:   serial
Err:   serial
SCSI:  Error: SCSI Controller(s) 1B4B:9170 not found
Net:   Fman1: Uploading microcode version 106.4.18
FM1@DTSEC1, FM1@DTSEC2, FM1@DTSEC3, FM1@DTSEC4, FM1@DTSEC5, FM1@DTSEC6, FM1@TGEC1
Hit any key to stop autoboot:  0
=> pri
baudrate=115200
bootargs=console=ttyS0,115200 root=/dev/ram0 earlycon=uart8250,0x21c0500,115200
bootcmd=cpld reset altbank
bootdelay=1
console=ttyAMA0,38400n8
eth1addr=00:e0:0c:00:77:01
eth2addr=00:e0:0c:00:77:02
eth3addr=00:e0:0c:00:77:03
eth4addr=00:e0:0c:00:77:04
eth5addr=00:e0:0c:00:77:05
eth6addr=00:e0:0c:00:77:06
ethact=FM1@DTSEC1
ethaddr=00:e0:0c:00:77:00
ethprime=e1000#0
fdt_high=0xffffffffffffffff
fdtaddr=8f000000
fdtcontroladdr=ffbc7738
fman_ucode=60300000
gatewayip=192.168.1.1
hwconfig=fsl_ddr:bank_intlv=auto
initrd_high=0xffffffffffffffff
ipaddr=10.193.20.234
kernel_addr=0x100000
kernel_load=0xa0000000
kernel_size=0x2800000
kernel_start=0x61100000
loadaddr=82000000
netmask=255.255.255.0
ramdisk_addr=0x800000
ramdisk_size=0x2000000
ramdiskaddr=88000000
scsidevs=0
serverip=10.192.208.142
stderr=serial
stdin=serial
stdout=serial
wrtboot=setenv bootargs root=/dev/mtdblock6 rootfstype=squashfs,jffs2 noinitrd earlycon=uart8250,mmio,0x21c0500 console=ttyS0,115200 mtdparts=60000000.nor:1M(rcw),1M(u-boot),1M(u-boot-env),1M(fman),1M(dtb),5M(kernel),54M(rootfs),64M(otherbank),60M@0x400000(firmware) && cp.b 60400000 $fdtaddr 100000 && cp.b 60500000 $loadaddr 500000 && bootm $loadaddr - $fdtaddr

Environment size: 1289/131068 bytes
=> run wrtboot
## Booting kernel from Legacy Image at 82000000 ...
   Image Name:   ARM64 OpenWrt Linux-4.4.14
   Image Type:   AArch64 Linux Kernel Image (gzip compressed)
   Data Size:    3238887 Bytes = 3.1 MiB
   Load Address: 80080000
   Entry Point:  80080000
   Verifying Checksum ... OK
## Flattened Device Tree blob at 8f000000
   Booting using the fdt blob at 0x8f000000
   Uncompressing Kernel Image ... OK
   Using Device Tree in place at 000000008f000000, end 000000008f0197c6
fsl-ppa: Bad firmware image (not a FIT image)
fsl-ppa: Bad firmware image (not a FIT image)

Starting kernel ...

[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 4.4.14 (jyt@BP) (gcc version 5.3.0 (OpenWrt GCC 5.3.0 49930) ) #1 SMP Fri Sep 2 17:08:07 UTC 2016
[    0.000000] Boot CPU: AArch64 Processor [410fd034]
[    0.000000] earlycon: Early serial console at MMIO 0x21c0500 (options '')
[    0.000000] bootconsole [uart0] enabled
[    0.000000] efi: Getting EFI parameters from FDT:
[    0.000000] efi: UEFI not found.
[    0.000000] Reserved memory: initialized node qman-fqd, compatible id fsl,qman-fqd
[    0.000000] Reserved memory: initialized node qman-pfdr, compatible id fsl,qman-pfdr
[    0.000000] Reserved memory: initialized node bman-fbpr, compatible id fsl,bman-fbpr
[    0.000000] PERCPU: Embedded 15 pages/cpu @ffffffc07fd8e000 s23808 r8192 d29440 u61440
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] CPU features: enabling workaround for ARM erratum 845719
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 515592
[    0.000000] Kernel command line: root=/dev/mtdblock6 rootfstype=squashfs,jffs2 noinitrd earlycon=uart8250,mmio,0x21c0500 console=ttyS0,115200 mtdparts=60000000.nor:1M(rcw),1M(u-boot),1M(u-boot-env),1M(fman),1M(dtb),5M(kernel),54M(rootfs),64M(otherbank),60M@0x400000(firmware)
[    0.000000] PID hash table entries: 4096 (order: 3, 32768 bytes)
[    0.000000] Dentry cache hash table entries: 262144 (order: 9, 2097152 bytes)
[    0.000000] Inode-cache hash table entries: 131072 (order: 8, 1048576 bytes)
[    0.000000] software IO TLB [mem 0xf6200000-0xfa200000] (64MB) mapped at [ffffffc076200000-ffffffc07a1fffff]
[    0.000000] Memory: 1927068K/2095104K available (4781K kernel code, 282K rwdata, 2120K rodata, 264K init, 811K bss, 168036K reserved, 0K cma-reserved)
[    0.000000] Virtual kernel memory layout:
[    0.000000]     vmalloc : 0xffffff8000000000 - 0xffffffbdbfff0000   (   246 GB)
[    0.000000]     vmemmap : 0xffffffbdc0000000 - 0xffffffbfc0000000   (     8 GB maximum)
[    0.000000]               0xffffffbdc0000000 - 0xffffffbdc1ff8000   (    31 MB actual)
[    0.000000]     fixed   : 0xffffffbffabfd000 - 0xffffffbffac00000   (    12 KB)
[    0.000000]     PCI I/O : 0xffffffbffae00000 - 0xffffffbffbe00000   (    16 MB)
[    0.000000]     modules : 0xffffffbffc000000 - 0xffffffc000000000   (    64 MB)
[    0.000000]     memory  : 0xffffffc000000000 - 0xffffffc07fe00000   (  2046 MB)
[    0.000000]       .init : 0xffffffc000740000 - 0xffffffc000782000   (   264 KB)
[    0.000000]       .text : 0xffffffc000080000 - 0xffffffc00073f4e4   (  6910 KB)
[    0.000000]       .data : 0xffffffc000792000 - 0xffffffc0007d8a00   (   283 KB)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] Hierarchical RCU implementation.
[    0.000000]  CONFIG_RCU_FANOUT set to non-default value of 32
[    0.000000] NR_IRQS:64 nr_irqs:64 0
[    0.000000] GIC: Using split EOI/Deactivate mode
[    0.000000] Architected cp15 timer(s) running at 25.00MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x5c40939b5, max_idle_ns: 440795202646 ns
[    0.000002] sched_clock: 56 bits at 25MHz, resolution 40ns, wraps every 4398046511100ns
[    0.008164] Calibrating delay loop (skipped), value calculated using timer frequency.. 50.00 BogoMIPS (lpj=250000)
[    0.018568] pid_max: default: 32768 minimum: 301
[    0.023237] Mount-cache hash table entries: 4096 (order: 3, 32768 bytes)
[    0.029972] Mountpoint-cache hash table entries: 4096 (order: 3, 32768 bytes)
[    0.037518] EFI services will not be available.
[    0.042083] ASID allocator initialised with 65536 entries
[    0.048012] Detected VIPT I-cache on CPU1
[    0.048030] CPU1: Booted secondary processor [410fd034]
[    0.048145] Detected VIPT I-cache on CPU2
[    0.048157] CPU2: Booted secondary processor [410fd034]
[    0.048265] Detected VIPT I-cache on CPU3
[    0.048277] CPU3: Booted secondary processor [410fd034]
[    0.048297] Brought up 4 CPUs
[    0.079067] SMP: Total of 4 processors activated.
[    0.083789] CPU: All CPU(s) started at EL2
[    0.087906] alternatives: patching kernel code
[    0.095133] DMI not present or invalid.
[    0.099088] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.109031] atomic64_test: passed
[    0.112721] NET: Registered protocol family 16
[    0.117574] fsl-mc bus type registered
[    0.121398] MC object device driver fsl_mc_dprc registered
[    0.126933] MC object device driver fsl_mc_allocator registered
[    0.132922] Bman ver:0a02,02,01
[    0.140386] qman-fqd addr 0xff000000 size 0x800000
[    0.145201] qman-pfdr addr 0xfc000000 size 0x2000000
[    0.150186] Qman ver:0a01,03,02,00
[    0.153672] vdso: 2 pages (1 code @ ffffffc000799000, 1 data @ ffffffc000798000)
[    0.161207] DMA: preallocated 256 KiB pool for atomic allocations
[    0.167364] Serial: AMBA PL011 UART driver
[    0.211862] ACPI: Interpreter disabled.
[    0.215938] SCSI subsystem initialized
[    0.219920] usbcore: registered new interface driver usbfs
[    0.225477] usbcore: registered new interface driver hub
[    0.230837] usbcore: registered new device driver usb
[    0.235982] dmi: Firmware registration failed.
[    0.240469] bman-fbpr addr 0xfe000000 size 0x1000000
[    0.245481] Bman err interrupt handler present
[    0.250351] Bman portal initialised, cpu 0
[    0.254522] Bman portal initialised, cpu 1
[    0.258695] Bman portal initialised, cpu 2
[    0.262870] Bman portal initialised, cpu 3
[    0.266978] Bman portals initialised
[    0.271432] Qman err interrupt handler present
[    0.276191] QMan: Allocated lookup table at ffffff800016d000, entry count 131073
[    0.284359] Qman portal initialised, cpu 0
[    0.288926] Qman portal initialised, cpu 1
[    0.293492] Qman portal initialised, cpu 2
[    0.298053] Qman portal initialised, cpu 3
[    0.302167] Qman portals initialised
[    0.305842] Bman: BPID allocator includes range 32:32
[    0.310951] Qman: FQID allocator includes range 256:256
[    0.316200] Qman: FQID allocator includes range 32768:32768
[    0.321837] Qman: CGRID allocator includes range 0:256
[    0.327115] Qman: pool channel allocator includes range 1025:15
[    0.333116] No USDPAA memory, no 'fsl,usdpaa-mem' in device-tree
[    0.339187] fsl-ifc 1530000.ifc: Freescale Integrated Flash Controller
[    0.345749] fsl-ifc 1530000.ifc: IFC version 1.4, 8 banks
[    0.351737] clocksource: Switched to clocksource arch_sys_counter
[    0.358002] pnp: PnP ACPI: disabled
[    0.362076] NET: Registered protocol family 2
[    0.366695] TCP established hash table entries: 16384 (order: 5, 131072 bytes)
[    0.374054] TCP bind hash table entries: 16384 (order: 6, 262144 bytes)
[    0.380883] TCP: Hash tables configured (established 16384 bind 16384)
[    0.387496] UDP hash table entries: 1024 (order: 3, 32768 bytes)
[    0.393561] UDP-Lite hash table entries: 1024 (order: 3, 32768 bytes)
[    0.400148] NET: Registered protocol family 1
[    0.404649] RPC: Registered named UNIX socket transport module.
[    0.410592] RPC: Registered udp transport module.
[    0.415318] RPC: Registered tcp transport module.
[    0.420038] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    0.427019] futex hash table entries: 1024 (order: 5, 131072 bytes)
[    0.436692] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.442935] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.452986] fuse init (API version 7.23)
[    0.457370] io scheduler noop registered (default)
[    0.462780] PCI host bridge /soc/pcie@3500000 ranges:
[    0.467860]    IO 0x4800010000..0x480001ffff -> 0x00000000
[    0.473376]   MEM 0x4840000000..0x487fffffff -> 0x40000000
[    0.478967] layerscape-pcie 3500000.pcie: PCI host bridge to bus 0000:00
[    0.485706] pci_bus 0000:00: root bus resource [bus 00-ff]
[    0.491214] pci_bus 0000:00: root bus resource [io  0x0000-0xffff]
[    0.497425] pci_bus 0000:00: root bus resource [mem 0x4840000000-0x487fffffff] (bus address [0x40000000-0x7fffffff])
[    0.521788] pci 0000:00:00.0: BAR 1: assigned [mem 0x4840000000-0x4843ffffff]
[    0.528958] pci 0000:00:00.0: BAR 0: assigned [mem 0x4844000000-0x4844ffffff]
[    0.536136] pci 0000:00:00.0: BAR 6: assigned [mem 0x4845000000-0x4845ffffff pref]
[    0.543746] pci 0000:00:00.0: BAR 8: assigned [mem 0x4846000000-0x48460fffff]
[    0.550916] pci 0000:01:00.0: BAR 0: assigned [mem 0x4846000000-0x484601ffff 64bit]
[    0.558649] pci 0000:01:00.0: BAR 6: assigned [mem 0x4846020000-0x484602ffff pref]
[    0.566258] pci 0000:00:00.0: PCI bridge to [bus 01]
[    0.571243] pci 0000:00:00.0:   bridge window [mem 0x4846000000-0x48460fffff]
[    0.578577] PCI host bridge /soc/pcie@3600000 ranges:
[    0.583658]    IO 0x5000010000..0x500001ffff -> 0x00000000
[    0.589167]   MEM 0x5040000000..0x507fffffff -> 0x40000000
[    0.594737] layerscape-pcie 3600000.pcie: PCI host bridge to bus 0001:00
[    0.601469] pci_bus 0001:00: root bus resource [bus 00-ff]
[    0.606984] pci_bus 0001:00: root bus resource [io  0x10000-0x1ffff] (bus address [0x0000-0xffff])
[    0.615992] pci_bus 0001:00: root bus resource [mem 0x5040000000-0x507fffffff] (bus address [0x40000000-0x7fffffff])
[    0.626709] pci 0001:00:00.0: bridge configuration invalid ([bus 00-00]), reconfiguring
[    0.634808] pci 0001:00:00.0: BAR 1: assigned [mem 0x5040000000-0x5043ffffff]
[    0.641983] pci 0001:00:00.0: BAR 0: assigned [mem 0x5044000000-0x5044ffffff]
[    0.649152] pci 0001:00:00.0: BAR 6: assigned [mem 0x5045000000-0x5045ffffff pref]
[    0.656761] pci 0001:00:00.0: PCI bridge to [bus 01]
[    0.661766] pcieport 0001:00:00.0: enabling device (0000 -> 0002)
[    0.668312] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    0.675133] console [ttyS0] disabled
[    0.678734] 21c0500.serial: ttyS0 at MMIO 0x21c0500 (irq = 18, base_baud = 25000000) is a 16550A
[    0.687579] console [ttyS0] enabled
[    0.687579] console [ttyS0] enabled
[    0.694556] bootconsole [uart0] disabled
[    0.694556] bootconsole [uart0] disabled
[    0.702514] 21c0600.serial: ttyS1 at MMIO 0x21c0600 (irq = 18, base_baud = 25000000) is a 16550A
[    0.711367] of_serial: probe of 21d0500.serial failed with error -28
[    0.717746] of_serial: probe of 21d0600.serial failed with error -28
[    0.724352] Unable to detect cache hierarchy from DT for CPU 0
[    0.733483] brd: module loaded
[    0.738529] loop: module loaded
[    0.743213] 60000000.nor: Found 1 x16 devices at 0x0 in 16-bit bank. Manufacturer ID 0x000089 Chip ID 0x00227e
[    0.753208] Amd/Fujitsu Extended Query Table at 0x0040
[    0.758343]   Amd/Fujitsu Extended Query version 1.3.
[    0.763389] number of CFI chips: 1
[    0.766790] 9 cmdlinepart partitions found on MTD device 60000000.nor
[    0.773222] Creating 9 MTD partitions on "60000000.nor":
[    0.778527] 0x000000000000-0x000000100000 : "rcw"
[    0.783423] ftl_cs: FTL header not found.
[    0.787621] 0x000000100000-0x000000200000 : "u-boot"
[    0.792759] ftl_cs: FTL header not found.
[    0.796937] 0x000000200000-0x000000300000 : "u-boot-env"
[    0.802424] ftl_cs: FTL header not found.
[    0.806617] 0x000000300000-0x000000400000 : "fman"
[    0.811572] ftl_cs: FTL header not found.
[    0.815757] 0x000000400000-0x000000500000 : "dtb"
[    0.820625] ftl_cs: FTL header not found.
[    0.824808] 0x000000500000-0x000000a00000 : "kernel"
[    0.829937] ftl_cs: FTL header not found.
[    0.834126] 0x000000a00000-0x000004000000 : "rootfs"
[    0.839257] ftl_cs: FTL header not found.
[    0.843436] mtd: device 6 (rootfs) set to be root filesystem
[    0.849096] 1 squashfs-split partitions found on MTD device rootfs
[    0.855271] 0x000000d80000-0x000004000000 : "rootfs_data"
[    0.860834] ftl_cs: FTL header not found.
[    0.865016] 0x000004000000-0x000008000000 : "otherbank"
[    0.870409] ftl_cs: FTL header not found.
[    0.874593] 0x000000400000-0x000004000000 : "firmware"
[    0.879896] ftl_cs: FTL header not found.
[    0.884614] nand: device found, Manufacturer ID: 0x2c, Chip ID: 0xac
[    0.890956] nand: Micron MT29F4G08ABBDAH4
[    0.894959] nand: 512 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    0.902783] Bad block table found at page 262080, version 0x01
[    0.909077] Bad block table found at page 262016, version 0x01
[    0.917207] ftl_cs: FTL header not found.
[    0.921391] fsl,ifc-nand 7e800000.nand: IFC NAND device at 0x7e800000, bank 1
[    0.928959] libphy: Fixed MDIO Bus: probed
[    0.933158] libphy: Freescale XGMAC MDIO Bus: probed
[    0.939449] libphy: Freescale XGMAC MDIO Bus: probed
[    0.945933] libphy: Freescale XGMAC MDIO Bus: probed
[    0.950931] libphy: Freescale XGMAC MDIO Bus: probed
[    0.955935] libphy: Freescale XGMAC MDIO Bus: probed
[    0.960931] libphy: Freescale XGMAC MDIO Bus: probed
[    0.965934] libphy: Freescale XGMAC MDIO Bus: probed
[    0.970934] libphy: Freescale XGMAC MDIO Bus: probed
[    0.975935] libphy: Freescale XGMAC MDIO Bus: probed
[    0.992425] Freescale FM module, FMD API version 21.1.0
[    0.999331] Freescale FM Ports module
[    1.002992] fsl_mac: fsl_mac: FSL FMan MAC API based driver
[    1.008897] fsl_mac 1ae0000.ethernet: FMan MEMAC
[    1.013515] fsl_mac 1ae0000.ethernet: FMan MAC address: 00:e0:0c:00:77:00
[    1.020580] fsl_mac 1ae2000.ethernet: FMan MEMAC
[    1.025195] fsl_mac 1ae2000.ethernet: FMan MAC address: 00:e0:0c:00:77:01
[    1.032023] fsl_mac 1ae4000.ethernet: FMan MEMAC
[    1.036632] fsl_mac 1ae4000.ethernet: FMan MAC address: 00:e0:0c:00:77:02
[    1.043462] fsl_mac 1ae6000.ethernet: FMan MEMAC
[    1.048072] fsl_mac 1ae6000.ethernet: FMan MAC address: 00:e0:0c:00:77:03
[    1.055143] fsl_mac 1ae8000.ethernet: FMan MEMAC
[    1.059752] fsl_mac 1ae8000.ethernet: FMan MAC address: 00:e0:0c:00:77:04
[    1.066823] fsl_mac 1aea000.ethernet: FMan MEMAC
[    1.071433] fsl_mac 1aea000.ethernet: FMan MAC address: 00:e0:0c:00:77:05
[    1.078266] fsl_mac 1af0000.ethernet: FMan MEMAC
[    1.082881] fsl_mac 1af0000.ethernet: FMan MAC address: 00:e0:0c:00:77:06
[    1.089735] fsl_dpa: FSL DPAA Ethernet driver
[    1.095812] fsl_dpa: fsl_dpa: Probed interface eth0
[    1.102195] fsl_dpa: fsl_dpa: Probed interface eth1
[    1.108920] fsl_dpa: fsl_dpa: Probed interface eth2
[    1.116015] fsl_dpa: fsl_dpa: Probed interface eth3
[    1.123475] fsl_dpa: fsl_dpa: Probed interface eth4
[    1.131286] fsl_dpa: fsl_dpa: Probed interface eth5
[    1.139470] fsl_dpa: fsl_dpa: Probed interface eth6
[    1.144400] fsl_advanced: FSL DPAA Advanced drivers:
[    1.149355] fsl_proxy: FSL DPAA Proxy initialization driver
[    1.155045] fsl_dpa_shared: FSL DPAA Shared Ethernet driver
[    1.160693] fsl_dpa_macless: FSL DPAA MACless Ethernet driver
[    1.166517] fsl_oh: FSL FMan Offline Parsing port driver
[    1.172657] Freescale USDPAA process driver
[    1.176829] fsl-usdpaa: no region found
[    1.180656] Freescale USDPAA process IRQ driver
[    1.185546] NET: Registered protocol family 10
[    1.190644] NET: Registered protocol family 17
[    1.195101] bridge: automatic filtering via arp/ip/ip6tables has been deprecated. Update your scripts to load br_netfilter if you need this.
[    1.207700] 8021q: 802.1Q VLAN Support v1.8
[    1.212242] fsl_generic: FSL DPAA Generic Ethernet driver
[    1.218208] fdt: not creating '/sys/firmware/fdt': CRC check failed
[    1.226374] VFS: Mounted root (squashfs filesystem) readonly on device 31:6.
[    1.233532] Freeing unused kernel memory: 264K (ffffffc000740000 - ffffffc000782000)
[    1.241275] Freeing alternatives memory: 44K (ffffffc000782000 - ffffffc00078d000)
[    1.588799] init: Console is alive
[    1.713986] pps_core: LinuxPPS API ver. 1 registered
[    1.718947] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it<mailto:giometti@linux.it>>
[    1.728407] PTP clock support registered
[    2.334736] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    2.340221] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 1
[    2.347954] xhci-hcd xhci-hcd.0.auto: hcc params 0x0220f66d hci version 0x100 quirks 0x00010010
[    2.356670] xhci-hcd xhci-hcd.0.auto: irq 27, io mem 0x02f00000
[    2.362800] hub 1-0:1.0: USB hub found
[    2.366549] hub 1-0:1.0: 1 port detected
[    2.370544] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    2.376029] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 2
[    2.383719] usb usb2: We don't know the algorithms for LPM for this host, disabling LPM.
[    2.391974] hub 2-0:1.0: USB hub found
[    2.395721] hub 2-0:1.0: 1 port detected
[    2.399729] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    2.405216] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 3
[    2.412937] xhci-hcd xhci-hcd.1.auto: hcc params 0x0220f66d hci version 0x100 quirks 0x00010010
[    2.421639] xhci-hcd xhci-hcd.1.auto: irq 28, io mem 0x03000000
[    2.427731] hub 3-0:1.0: USB hub found
[    2.431477] hub 3-0:1.0: 1 port detected
[    2.435472] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    2.440950] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 4
[    2.448623] usb usb4: We don't know the algorithms for LPM for this host, disabling LPM.
[    2.456874] hub 4-0:1.0: USB hub found
[    2.460621] hub 4-0:1.0: 1 port detected
[    2.464654] xhci-hcd xhci-hcd.2.auto: xHCI Host Controller
[    2.470135] xhci-hcd xhci-hcd.2.auto: new USB bus registered, assigned bus number 5
[    2.477856] xhci-hcd xhci-hcd.2.auto: hcc params 0x0220f66d hci version 0x100 quirks 0x00010010
[    2.486563] xhci-hcd xhci-hcd.2.auto: irq 29, io mem 0x03100000
[    2.492667] hub 5-0:1.0: USB hub found
[    2.496414] hub 5-0:1.0: 1 port detected
[    2.500405] xhci-hcd xhci-hcd.2.auto: xHCI Host Controller
[    2.505890] xhci-hcd xhci-hcd.2.auto: new USB bus registered, assigned bus number 6
[    2.513562] usb usb6: We don't know the algorithms for LPM for this host, disabling LPM.
[    2.521821] hub 6-0:1.0: USB hub found
[    2.525568] hub 6-0:1.0: 1 port detected
[    2.531036] sdhci: Secure Digital Host Controller Interface driver
[    2.537217] sdhci: Copyright(c) Pierre Ossman
[    2.541773] sdhci-pltfm: SDHCI platform and OF driver helper
[    2.548025] usbcore: registered new interface driver usb-storage
[    2.555690] sdhci-esdhc 1560000.esdhc: No vmmc regulator found
[    2.561511] sdhci-esdhc 1560000.esdhc: No vqmmc regulator found
[    2.611744] mmc0: SDHCI controller on 1560000.esdhc [1560000.esdhc] using ADMA 64-bit
[    2.624455] init: - preinit -
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    2.811756] usb usb5-port1: over-current condition
[    2.841750] usb usb6-port1: over-current condition
[    5.743965] jffs2: notice: (850) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
[    5.759779] mount_root: switching to jffs2 overlay
[    5.767657] procd: - early -
[    6.382413] procd: - ubus -
[    6.434799] random: ubusd urandom read with 4 bits of entropy available
[    6.441685] procd: - init -
Please press Enter to activate this console.
[    6.508755] i2c /dev entries driver
[    6.512984] i2c i2c-0: IMX I2C adapter registered
[    6.517680] i2c i2c-0: can't use DMA
[    6.522801] e1000e: Intel(R) PRO/1000 Network Driver - 3.2.6-k
[    6.528625] e1000e: Copyright(c) 1999 - 2015 Intel Corporation.
[    7.992255] device eth0 entered promiscuous mode
[    7.997580] br-lan: port 1(eth0) entered forwarding state
[    8.002988] br-lan: port 1(eth0) entered forwarding state
[    9.992204] br-lan: port 1(eth0) entered disabled state



BusyBox v1.24.2 () built-in shell (ash)

  _______                     ________        __
|       |.-----.-----.-----.|  |  |  |.----.|  |_
|   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
|_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
-----------------------------------------------------
DESIGNATED DRIVER (Bleeding Edge, 49929)
-----------------------------------------------------
  * 2 oz. Orange Juice         Combine all juices in a
  * 2 oz. Pineapple Juice      tall glass filled with
  * 2 oz. Grapefruit Juice     ice, stir well.
  * 2 oz. Cranberry Juice
-----------------------------------------------------
root@OpenWrt:/# ifconfig
br-lan    Link encap:Ethernet  HWaddr 00:E0:0C:00:77:00
          inet addr:192.168.1.1  Bcast:192.168.1.255  Mask:255.255.255.0
          inet6 addr: fd67:8579:a0a8::1/60 Scope:Global
          inet6 addr: fe80::2e0:cff:fe00:7700/64 Scope:Link
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:8 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0 (0.0 B)  TX bytes:896 (896.0 B)

eth0      Link encap:Ethernet  HWaddr 00:E0:0C:00:77:00
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:9 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0 (0.0 B)  TX bytes:942 (942.0 B)
          Memory:1ae0000-1ae0fff

eth1      Link encap:Ethernet  HWaddr 00:E0:0C:00:77:01
          inet addr:10.193.20.43  Bcast:10.193.20.255  Mask:255.255.255.0
          inet6 addr: fe80::2e0:cff:fe00:7701/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:49 errors:0 dropped:0 overruns:0 frame:0
          TX packets:10 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:4172 (4.0 KiB)  TX bytes:1844 (1.8 KiB)
          Memory:1ae2000-1ae2fff

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:8 errors:0 dropped:0 overruns:0 frame:0
          TX packets:8 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:656 (656.0 B)  TX bytes:656 (656.0 B)

root@OpenWrt:/# ping 10.193.20.106
PING 10.193.20.106 (10.193.20.106): 56 data bytes
64 bytes from 10.193.20.106: seq=0 ttl=64 time=12.728 ms
64 bytes from 10.193.20.106: seq=1 ttl=64 time=0.169 ms
^C
--- 10.193.20.106 ping statistics ---
2 packets transmitted, 2 packets received, 0% packet loss
round-trip min/avg/max = 0.169/6.448/12.728 ms
root@OpenWrt:/#

From: Zoltan Herpai [mailto:notifications@github.com]
Sent: Saturday, September 03, 2016 4:40 PM
To: openwrt/openwrt
Cc: Yutang Jiang; Mention
Subject: Re: [openwrt/openwrt] [v3]layerscape: add 64b/32b target for ls1043ardb device (#81)


@fsl-jyt<https://github.com/fsl-jyt> - could you please verify if it compiles and runs fine with 4.4.14 which is currently in OpenWrt trunk? Same goes for the image building.

Thanks,
Zoltan H

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/openwrt/openwrt/pull/81#issuecomment-244534852>, or mute the thread<https://github.com/notifications/unsubscribe-auth/APKzf-Mj4BpuCF1o1mU5HO9UZwXUtzFTks5qmTJ3gaJpZM4JyW3Q>.
